Applications/Spire/Include/Spire/Charting/ChartView.hpp

```diff
@@ -14,6 +14,52 @@ namespace Spire {
   class ChartView : public QWidget {
     public:
 
+      //! Represents a region defined by two points.
+      struct Region {
+
+        /** The top left point of the region. */
+        ChartPoint m_top_left;
+
+        /** The bottom right point of the region. */
+        ChartPoint m_bottom_right;
+      };
+
+      //! Represents a non-contiguous gap between two ChartValues.
+      struct Gap {
+
+        /** The start value of the gap. */
+        ChartValue m_start;
+
+        /** The end value of the gap. */
+        ChartValue m_end;
+      };
+
+      //! Represents an accumulation of loaded data and gaps, and stores values
+      //! that determine what to load next.
+      struct LoadedData {
```

Comment:
Why are these ```struct```s public? No changes to the public API should be made without first proposing them.

Reply:
Is there a way to make it accessible to the anonymous namespace and usable as a private value type member?

Reply:
No, you will need to use a private static method.

Reply:
Do I declare static members alphabetically with the other private methods, or does it go before or after them?

Reply:
Good question, static methods go just above non-static methods.